framework/di/Container.php

```diff
@@ -456,48 +457,68 @@ protected function resolveDependencies($dependencies, $reflection = null)
     }
 
     /**
-     * Invoke callback with resolved dependecies parameters.
+     * Resolve dependencies for a function, can use either named or numbered $params.
      * @param callable $callback callable to be invoked.
      * @param array $params callback paramater.
-     * @return mixed the callback return value.
+     * @return array The resolved dependencies.
      * @throws InvalidConfigException if a dependency cannot be resolved or if a dependency cannot be fulfilled.
      * @since 2.0.7
      */
-    public function invoke($callback, $params = [])
+    public function resolve(callable $callback, $params = [])
```

Comment:
Visibility is a question, yes. @SamMousa what's the use case for making it public?